### PR TITLE
feat: implement BackfillAsyncOperations and improve async operation reliability

### PR DESCRIPTION
**Pull request by jessejlt**
## Summary

This PR implements the `BackfillAsyncOperations` method for SummaryRule CRDs and refactors async operation management to improve reliability during Kusto outages.

### 1. Implement BackfillAsyncOperations Method (`api/v1/summaryrule_types.go`)

- **New Method**: `BackfillAsyncOperations(clk clock.Clock)` that automatically generates missing async operations for time intervals between the last successful execution and current time
- **Time Window Logic**: Calculates proper execution windows accounting for ingestion delay and interval alignment
- **Duplicate Prevention**: Uses time window-based deduplication to avoid creating duplicate operations
- **Capacity Management**: Maintains the 200-operation limit by pruning oldest operations chronologically
- **Backlog Operations**: Creates operations without `OperationId` for later submission when Kusto becomes available

### 2. Comprehensive Test Coverage (`api/v1/summaryrule_types_test.go`)

- **12 Test Cases** covering various scenarios:
  - No last execution time (no-op behavior)
  - Single and multiple missing intervals
  - Duplicate prevention logic
  - Capacity limit enforcement (200 operations)
  - Ingestion delay handling
  - Edge cases with existing operations

### 3. Conservative Async Operation Management (`ingestor/adx/tasks.go`)

- **Removed**: Aggressive stale operation pruning that could lose track of operations during outages
- **Simplified**: `trackAsyncOperations` method now skips operations not found in Kusto rather than removing them
- **Reliability**: Operations are only removed when definitively completed, preventing data loss

### 4. Test Cleanup (`ingestor/adx/tasks_test.go`)

- **Removed**: 3 obsolete tests that validated the old aggressive pruning behavior
- **Aligned**: Remaining tests with the new conservative approach

### Reliability Improvements
- **No Operation Loss**: Conservative approach prevents losing track of operations during Kusto outages
- **Automatic Recovery**: BackfillAsyncOperations ensures missing intervals are automatically detected and queued
- **Graceful Degradation**: System continues to function even when Kusto is temporarily unavailable

### Operational Excellence
- **Simplified Logic**: Fewer edge cases and race conditions in async operation handling
- **Deterministic Behavior**: Predictable operation management regardless of Kusto availability
- **Comprehensive Testing**: Extensive test coverage ensures reliability across various scenarios

### BackfillAsyncOperations Algorithm
1. **Input Validation**: Returns early if no last execution time exists
2. **Window Calculation**: Generates time windows from last execution forward using interval duration
3. **Boundary Checking**: Only creates operations for completely finished intervals (before current time - ingestion delay)
4. **Deduplication**: Uses `StartTime:EndTime` key mapping to prevent duplicate operations
5. **Capacity Management**: Chronologically sorts and prunes to maintain 200-operation limit
6. **Storage**: Updates the CRD condition with the new operation list

### Conservative Operation Management
- **Philosophy Change**: From "assume stale and remove" to "keep until definitively completed"
- **Kusto Integration**: Only queries Kusto for operation status, doesn't assume operations are stale if query fails
- **Error Handling**: Treats Kusto unavailability as temporary condition rather than operation failure

## Testing

All existing tests pass, plus 12 new comprehensive test cases for BackfillAsyncOperations covering:
- Edge cases (no execution time, no operations)
- Normal operations (single/multiple missing intervals)
- Capacity limits and pruning behavior
- Integration with existing async operation methods

```bash
go test ./api/v1 -v
go test ./ingestor/adx -v
```

## Breaking Changes

None. This is a purely additive feature with internal refactoring that maintains API compatibility.

## Files Changed

- `api/v1/summaryrule_types.go`: Added BackfillAsyncOperations method
- `api/v1/summaryrule_types_test.go`: Added comprehensive test suite
- `ingestor/adx/tasks.go`: Simplified async operation tracking logic  
- `ingestor/adx/tasks_test.go`: Removed obsolete tests for old behavior

### CODE REVIEW

File: api/v1/summaryrule_types.go [hunk 1]
```diff
@@ -318,6 +319,118 @@ func (s *SummaryRule) NextExecutionWindow(clk clock.Clock) (windowStartTime time
...
+
+	// Marshal and store the updated operations
+	operationsJSON, err := json.Marshal(allOperations)
+	if err != nil {
+		// If we can't marshal the JSON, something has gone horribly wrong
+		return
+	}
+
+	// Update the condition
+	condition := meta.FindStatusCondition(s.Status.Conditions, SummaryRuleOperationIdOwner)
```
[the remark this answers is not in the document]
Comment (jessejlt) [reply api/v1/summaryrule_types.go:420]: `meta.SetStatusCondition` only actually updates the subresource's status if something changes in the condition, so I've updated the code to set a more reasonable status and reason based on the content, but left SetStatusCondition in place so that if something does change, like in the Generation, then the CRD can be updated.

**Review by matucker-msft — APPROVED** [review]
